src/main/java/gift/service/ProductService.java

```diff
@@ -56,4 +60,41 @@ public Page<Product> findAll(Pageable pageable) {
     public void delete(Long id) {
         productRepository.deleteById(id);
     }
+
+    // 옵션
+    public List<Option> findOptionsByProductId(Long productId) {
+        return optionRepository.findByProductId(productId);
+    }
+
+    public Option findOptionById(Long id) {
+        return optionRepository.findById(id)
+                .orElseThrow(() -> new IllegalArgumentException("옵션을 찾을 수 없습니다."));
+    }
+
+    @Transactional
+    public void updateProduct(Long id, Product product) {
+        Product existing = productRepository.findById(id).orElseThrow();
+
+        // 옵션 먼저 삭제
+        optionRepository.deleteByProductId(id);
+
+        // 기존 상품 정보 업데이트
+        existing.setName(product.getName());
+        existing.setPrice(product.getPrice());
+        existing.setImageUrl(product.getImageUrl());
+        existing.setMDapproved(product.getMDapproved());
+
+        // 새 옵션 세팅
+        if (product.getOptions() != null) {
+            for (Option opt : product.getOptions()) {
+                opt.setProduct(existing); // 양방향 연관
+            }
+            existing.getOptions().clear(); // 리스트 클리어
+            existing.getOptions().addAll(product.getOptions());
+        }
+
+        productRepository.save(existing);
+    }
```

Comment:
```suggestion
@Transactional
public void updateProduct(Long id, Product updatedProduct) {
    Product existing = productRepository.findById(id)
            .orElseThrow(() -> new ProductNotFoundException("상품을 찾을 수 없습니다."));
    
    existing.updateBasicInfo(updatedProduct);
    
    //  cascade + orphanRemoval로 안전하게 처리
    existing.replaceOptions(updatedProduct.getOptions());   
    existing.validateBusinessRules();
}
```
- 이 부분도 개선해보세요. JPA Dirty Checking을 적극적으로 활용해보시구요. 지금과 같이 사용하는건 JPA 관점에서는 조금 어색합니다. 가령, existing를 조회해온 시점에 Product와 Options가 영속성 컨텍스트에 로드되었을텐데요. 이를 `optionRepository.deleteByProductId(id);` 로 삭제한다 한들 커밋이 이루어지지 않았으니 아직 영속성 컨텍스트에는 존재하게 됩니다. 상황에 따라 이렇게 작성하는 경우들도 있기는 하지만, 현재 맥락에서는 오히려 휴먼에러가 생길 여지가 있어보여요. 중간에 자칫 커밋되어 버리면, 이 후 영속성 컨텍스트 내의 컬렉션 조작하는 로직들에서 예상치 못한 결과가 나올 수도 있구요.

Reply:
- JPA의 경우에 영속성 컨텍스트에 정보를 DB 조회 시, Product의 정보를 저장하고, 이를 JPA가 추적하기 때문에 변경 사항이 발생하면 Dirty Checking을 통해 알아서 update를 해줍니다. 라고 주신 피드백의 내용을 이해하였습니다. 혹시 제가 잘못 이해한 부분이 있을 시 지적부탁드리겠습니다.  
- Product에 updateBasicInfo과 위에서 주셨던 피드백을 하면서 만든 setOptions(replaceOptions)로 구성하였습니다. 나머지 validateBusinessRules의 경우에는 어떤 조건을 걸어야 할 지 아직 떠오르지 않아 추가하지 않았습니다.

Reply:
- 예시 코드는 피드백에 대한 이해를 돕고자 작성하는 뼈대 코드일 뿐, 저대로 작성하라는 의미가 아니에요~~~